synapse/app/synctl.py

```diff
@@ -190,7 +218,11 @@ def main():
         if start_stop_synapse:
             stop(pidfile, "synapse.app.homeserver")
 
-        # TODO: Wait for synapse to actually shutdown before starting it again
+    # Wait for synapse to actually shutdown before starting it again
+    if action == "restart" and os.path.exists(pidfile):
```

Comment:
start_stop_synapse isn't being toggled to `false` after it has executed the stop section `if action == "stop" or action == "restart":` 
So we can assume that it is `True` .
I could first toggle it to `False` then check if it is `False` in the _**waiting to restart**_ section then toggle it back so that the _**start section**_ runs fine .
But then I'll have had added 3 extra lines unnecessarily .
And I think you wouldn't want that 😛